File: classfields/api-tests/api-client/src/main/java/ru/auto/tests/moisha/ra/ResponseSpecBuilders.java
package ru.auto.tests.moisha.ra;

import io.restassured.builder.ResponseSpecBuilder;

import static org.hamcrest.CoreMatchers.equalTo;
import static ru.auto.tests.commons.restassured.ResponseSpecBuilders.shouldBe200OkJSON;

/**
 * Created by vicdev on 18.09.17.
 */
public class ResponseSpecBuilders {

    private ResponseSpecBuilders() {
    }

    public static ResponseSpecBuilder shouldBeEmptyJson() {
        return shouldBe200OkJSON().expectBody(equalTo("{}"));
    }
}

File: classfields/api-tests/salesman-api-client/src/main/scala/ru/auto/tests/ra/ResponseSpecBuilders.scala
package ru.auto.tests.ra

import io.restassured.builder.ResponseSpecBuilder
import org.apache.http.HttpStatus.SC_BAD_REQUEST
import org.hamcrest.CoreMatchers.equalTo
import ru.auto.tests.commons.restassured.ResponseSpecBuilders.{
  shouldBe200OkJSON,
  shouldBeCode
}

object ResponseSpecBuilders {

  def shouldBe400WithMissedSalesmanUser: ResponseSpecBuilder =
    shouldBeCode(SC_BAD_REQUEST).expectBody(
      equalTo("Request is missing required HTTP header 'X-Salesman-User'")
    )

  def shouldBeEmptyJson: ResponseSpecBuilder =
    shouldBe200OkJSON.expectBody(equalTo("{}"))

  def shouldBe200WithMessageOK: ResponseSpecBuilder =
    shouldBe200OkJSON.expectBody("message", equalTo("OK"))
}
